stdlib/src/builtin/float_literal.mojo

```diff
@@ -239,7 +239,38 @@ struct FloatLiteral(
             return truncated
         return truncated + 1
 
-    # TODO: implement __round__
+    @always_inline("nodebug")
+    fn __round__(self, ndigits: IntLiteral = 0) -> Self:
+        """Return the rounded value of the FloatLiteral.
+
+        Args:
+            ndigits: The number of digits to round to. Defaults to 0.
+        Returns:
+            The rounded value.
+        """
+        # Handle special values first.
+        if not self._is_normal():
+            return self
+
+        alias one = __mlir_attr.`#kgen.int_literal<1> : !kgen.int_literal`
+        alias ten = __mlir_attr.`#kgen.int_literal<10> : !kgen.int_literal`
+        var multiplier = one
+        for _ in range(ndigits):
+            multiplier = __mlir_op.`kgen.int_literal.binop`[
+                oper = __mlir_attr.`#kgen<int_literal.binop_kind mul>`
+            ](multiplier, ten)
+        var target: Self = self * Self(multiplier)  # 119.8
+        var truncated: IntLiteral = target.__int_literal__()  # 119.0
```

Comment:
To be defensive here (because implicit conversions can cause some unintuitive behavior below) and decrease dependency on `IntLiteral`, can you just declare this as `Self, and convert right away? Then you don't need the explicit casts to `Self` below.